compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirConstDeclarationChecker.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.declaration
+
+import org.jetbrains.kotlin.fir.FirFakeSourceElementKind
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.declarations.*
+import org.jetbrains.kotlin.lexer.KtTokens
+
+object FirConstDeclarationChecker : FirBasicDeclarationChecker() {
+    override fun check(declaration: FirDeclaration, context: CheckerContext, reporter: DiagnosticReporter) {
+        if (declaration !is FirVariable<*>) return
+        val source = declaration.source ?: return
+        if (source.kind is FirFakeSourceElementKind) return
```

Comment:
The first error is a different diagnostic `CONST_VAL_NOT_TOP_LEVEL_OR_OBJECT` which is not yet implemented. Removing this for now though seems harmless. I'll update.

The second error is `WRONG_MODIFIER_TARGET` but will be added later.

Reply:
As discussed offline, `CONST_VAL_NOT_TOP_LEVEL_OR_OBJECT` can be diagnosed with just the value parameter. There shouldn't be any need to check anything in the property with fake source. So I'll leave this check for `FirFakeSourceElementKind` here and in the unlikely event, it can be removed.